### PR TITLE
93.get lessons by teacher

### PR DESCRIPTION
**Pull request by Redingston**

### CODE REVIEW

File: src/YPS.Application/Lessons/Queries/GetLessonsByTeacher/LessonByTeacherDto.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        public string Auditorium { get; set; }
+        public long AuditoriumNumber { get; set; }
+        public long DisciplineId { get; set; }
+        public string Discipline { get; set; }
+        public long ClassId { get; set; }
+        public string ClassName { get; set; }
+        public long TeacherId { get; set; }
+
+
+        public void Mapping(Profile profile)
```
Comment (DmMov) [inline src/YPS.Application/Lessons/Queries/GetLessonsByTeacher/LessonByTeacherDto.cs]: Dorou, Sorry)

**Review by vunt1k — DISMISSED** [review]

**Review by DmMov — DISMISSED** [review]

**Review by vunt1k — DISMISSED** [review]

File: src/YPS.Application/Lessons/Queries/GetLessonsByTeacher/LessonByTeacherDto.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public long ClassId { get; set; }
+        public string ClassName { get; set; }
+        public long TeacherId { get; set; }
+
+        public void Mapping(Profile profile)
+        {
+            profile.CreateMap<Lesson, LessonByTeacherDto>()
+                .ForMember(e => e.LessonDate, opt => opt.MapFrom(e => e.LessonDate.ToString("dd.MM.yyyy")))
+                .ForMember(e => e.Auditorium, opt => opt.MapFrom(e => e.Auditorium.Name))
+                .ForMember(e=>e.AuditoriumNumber, opt=>opt.MapFrom(e =>e.Auditorium.Number))
```
Comment (bodkia22) [inline src/YPS.Application/Lessons/Queries/GetLessonsByTeacher/LessonByTeacherDto.cs]: It's a big problem

File: src/YPS.WebUI/Controllers/LessonsController.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using YPS.Application.Lessons.Queries.GetLessonsByTeacher;
+
+namespace YPS.WebUI.Controllers
+{
+    [Authorize(Roles ="teacher")]
```
Comment (bodkia22) [inline src/YPS.WebUI/Controllers/LessonsController.cs]: Put it to the querry

Comment (bodkia22) [reply src/YPS.WebUI/Controllers/LessonsController.cs]: And missed the 'probel'

**Review by bodkia22 — DISMISSED** [review]

**Review by DmMov — DISMISSED** [review]

File: src/YPS.Application/Lessons/Queries/GetLessonsByTeacher/LessonByTeacherDto.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        public long ClassId { get; set; }
+        public string ClassName { get; set; }
+        public long TeacherId { get; set; }
+
+        public void Mapping(Profile profile)
+        {
+            profile.CreateMap<Lesson, LessonByTeacherDto>()
+                .ForMember(e => e.LessonDate, opt => opt.MapFrom(e => e.LessonDate.ToString("dd.MM.yyyy")))
+                .ForMember(e => e.Auditorium, opt => opt.MapFrom(e => e.Auditorium.Name))
+                .ForMember(e => e.AuditoriumNumber, opt=>opt.MapFrom(e =>e.Auditorium.Number))
```
Comment (bodkia22) [inline src/YPS.Application/Lessons/Queries/GetLessonsByTeacher/LessonByTeacherDto.cs]: U know what to do

**Review by DmMov — APPROVED** [review]

**Review by vunt1k — APPROVED** [review]

**Review by bodkia22 — APPROVED** [review]

**Pull request merged**